lib/skate_web/auth_manager/error_handler.ex

```diff
@@ -1,10 +1,22 @@
 defmodule SkateWeb.AuthManager.ErrorHandler do
+  import Plug.Conn
+
   @behaviour Guardian.Plug.ErrorHandler
 
   @impl Guardian.Plug.ErrorHandler
   def auth_error(conn, {_type, _reason}, _opts) do
-    Phoenix.Controller.redirect(conn,
-      to: SkateWeb.Router.Helpers.auth_path(conn, :request, "cognito")
-    )
+    auth_retries = get_session(conn, :auth_retries) || 3
+
+    if auth_retries > 0 do
+      conn
+      |> put_session(:auth_retries, auth_retries - 1)
+      |> Phoenix.Controller.redirect(
+        to: SkateWeb.Router.Helpers.auth_path(conn, :request, "cognito")
+      )
+    else
+      conn
+      |> delete_session(:auth_retries)
+      |> send_resp(:unauthorized, "unauthorized")
+    end
```

Comment:
question: I don't see anything in splunk but I might not be searching for the right term, are there any logs for this so we can track this happening?

Reply:
If you search for `path=/auth/cognito/callback error` you can find instances where the `AuthController` received an error from Cognito and therefore triggered a redirect back to the start of the flow, is that the sort of thing you're looking for?

If we want to find instances of us reaching the max retries and giving the unauthorized response you could probably search for `controller=SkateWeb.PageController status=401` or something like that.